apps/automation-v2/src/tlc/batch-liquidate.ts

```diff
@@ -0,0 +1,217 @@
+  WALLET_NAME: string;
+  TLC_ADDRESS: string;
+  ACC_LIQ_MAX_CHUNK_NO: number;
+  MIN_ETH_BALANCE: bigint;
+}
+
+export async function batchLiquidate(
+  ctx: TaskContext,
+  config: TlcBatchLiquidateConfig
+): Promise<TaskResult> {
+  const provider = await ctx.getProvider(config.CHAIN.id);
+  const signer = await ctx.getSigner(provider, config.WALLET_NAME);
+  const walletAddress = await signer.getAddress();
+  const webhookUrl = await ctx.getSecret(DISCORD_WEBHOOK_URL_KEY);
+  const discord = await connectDiscord(webhookUrl, ctx.logger);
+
+  const tlc: ITempleLineOfCredit = ITempleLineOfCredit__factory.connect(
+    await config.TLC_ADDRESS,
+    signer
+  );
+
+  const chunkify = function (itr: string[], size: number) {
+    const chunk: string[][] = [];
+    let innerChunk: string[] = [];
+    for (const v of itr) {
+      innerChunk.push(v);
+      if (innerChunk.length === size) {
+        chunk.push(innerChunk);
+        innerChunk = [];
+      }
+    }
+    if (innerChunk.length) chunk.push(innerChunk);
+    return chunk;
+  };
+
+  // TODO: debug function, delete before merging
+  const checkAccPosition = async (accounts: string[]) => {
+    accounts.forEach(async (a) => {
+      ctx.logger.info(`**Acc to check ${a} **`);
+      const position = await tlc.accountPosition(a);
+      
+      ctx.logger.info(`Position: ${JSON.stringify({
+        collateral: formatUnits(position.collateral, 18),
+        currentDebt: formatUnits(position.currentDebt, 18),
+        maxBorrow: formatUnits(position.maxBorrow, 18),
+        healthFactor: formatUnits(position.healthFactor, 18),
+        loanToValueRatio: formatUnits(position.loanToValueRatio, 18),
+      })}`)
+
+    });
+  };
+
+  const submittedAt = new Date();
+
+  const url = getChainById(config.CHAIN.id).subgraphUrl;
+  const res = await getTlcUsers(url);
+  const tlcUsers = (await res()).data?.users;
+  // if undefined or zero users returned from subgraph, success silently
+  if (!tlcUsers || tlcUsers.length === 0) return taskSuccessSilent();
+  
+  const accountsToCheck = tlcUsers.flatMap((u) => u.id);
+  if (accountsToCheck.length === 0) return taskSuccessSilent();
+
+  // only liquidate the accounts which have exceeded the max ltv
+  const compLiquidityAccs = await tlc.computeLiquidity(accountsToCheck);
+  const accsToLiquidate: Array<string> = [];
+  compLiquidityAccs.map((acc, i) => {
+    if (acc.hasExceededMaxLtv) accsToLiquidate.push(accountsToCheck[i]);
+  });
+
+  if (accsToLiquidate.length === 0) return taskSuccessSilent();
+
+  // chunk compLiquidityAccs to a max number of requests per batchLiquidate
+  // e.g. try to liquidate 1000 accounts at once could use too much gas and fail
+  const accListChunks = chunkify(accsToLiquidate, config.ACC_LIQ_MAX_CHUNK_NO);
+  accListChunks.forEach(async (accBatch) => {
+    await checkAccPosition(accBatch); // check accs position before liquidation
+    const tx = await tlc.batchLiquidate(accBatch, {
+      gasLimit: 1_000_000n,
```

Comment:
Have you tested that we can liquidate 100 users for 1mm gas? I'm not sure we could.

Your test script might need to setup 100 accounts and try and liquidate them all

Reply:
Seems 1mm is enough, however I've added gasLimit as task config
```
╰─➤  yarn local-fork:mainnet-test:tlc
yarn run v1.19.0
$ (cd ../../protocol && forge script --fork-url http://127.0.0.1:8545 scripts/deploys/localhost/v2/999-localhost-tlc-mainnet-test.s.sol)
[⠢] Compiling...
[⠒] Compiling 1 files with 0.8.19
[⠑] Solc 0.8.19 finished in 1.32s
Compiler run successful!
Script ran successfully.

== Logs ==
  No of accounts to be liquidated 1
        Running tlc.batchLiquidate...
        -total gas used           144227
        -totalCollateralClaimed   1244800000000000000
        -totalDaiDebtWiped        1065551226618879322
  
  *Move forward 1 day to trigger liquidations*

  No of accounts to be liquidated 100
        Running tlc.batchLiquidate...
        -total gas used           681405
        -totalCollateralClaimed   500000000000000000000000
        -totalDaiDebtWiped        428069788581088020488000
✨  Done in 6.38s.
```